### ISSUE TITLE
Timezones issue

### ISSUE THREAD

**Issue by divan** [luisfarzati/ng-bs-daterangepicker#78]
When selecting range, timestamps seems to be in local timezone, but stored without TZ info (which means to be UTC)

I.e. is range "2015-08-01 - 2015-08-05" and local time zone GMT-04, timestamps should be '2015-08-01T04:00:00.000Z-04:00' - '2015-08-01T23:59:59.999Z-04:00' or in UTC '2015-08-01T04:00:00.000Z' - '2015-08-05T23:59:59.999Z'. (we need to get *dates* range, after all, regardless of browser timezone).

Instead model object contains '2015-08-01T04:00:00.000Z' - '2015-08-06T03:59:59.999Z' - without TZ info, which stands for UTC, which is incorrect. And any code (backend, for example) which need to deal with this range doesn't know what real timezone was and what real date range was meant.

Is it easy to fix, or it's some design flaw and not trivial to fix?

**Comment by alanjames1987** [comment]
I've always dealt with timezones client side before passing the dates to ui modules. This is because there is so much work in dealing with timezones and every company I have worked for has dealt with them differently.

I will refer you to this video to hopefully entertain while also talking illustrating why dealing with timezones is a pain.

https://www.youtube.com/watch?v=-5wpm-gesOY

If you have any suggestions about how this module should deal with them please suggest. I am more than willing to add something if it is warranted.

**Comment by divan** [comment]
You're right, timezones are tricky, and that's, actually, the point in case of *date*rangepicker - we don't need timezones at all here. Correct me if I'm wrong, but use case for date range picker is to handle date ranges - say, when you book a hotel. We don't work with exact timestamps (well, maybe check-in/check-out times need to be specified, but that's more of application level logic).

If I select range "Sep 10 - Oct 10 ", I want exactly this date range to be stored and shown to all clients, regardless of their timezone. That fact that 'Sep 10 00:00:00' in my timezone is 'Sep 09 18:00:00' in another timezone should be irrelevant to date range picker.

My suggestion would be to always store startDate and endDate as UTC with 00:00:00 time, but it would be backward incompatible with people who use it already. So I'd suggest to make new option, something like 'utc' or 'forceUTC' and implement it this way, stepping out of time/timezone calculations completely and let user deal with their particular case.

Currently I use this ugly code to get 'correct' date timestamp:
```javascript
var end = $scope.dateRange.endDate.toDate();
year = 1900 + end.getYear();
month = end.getMonth();
day = end.getDate();
var end_date = new Date(Date.UTC(year, month, day, 12, 00, 00));
```
But it's kinda sad :)
